examples/joystick/Joystick.cpp

```diff
@@ -0,0 +1,238 @@
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Graphics.hpp>
+#include <algorithm>
+#include <sstream>
+#include <iomanip>
+#include <string>
+#include <map>
+
+
+namespace
+{
+    struct JoystickObject
+    {
+        sf::Text label;
+        sf::Text value;
+    };
+
+    typedef std::map<std::string, JoystickObject> Texts;
+    Texts texts;
+    std::ostringstream sstr;
```

Comment:
What is the motivation for this global ss?

Reply:
Having to construct/destroy the ss everywhere where it is currently used would cause unnecessary overhead. I would have liked to use `std::to_string` but we aren't there yet. 😉

Reply:
> Having to construct/destroy the ss everywhere where it is currently used would cause unnecessary overhead.

Is it something that you measured? I've always seen people instantiating (or recommending to instantiate) this locally, only where it is needed.

Reply:
I always use stringstreams to stringify numerical values, also when getting a rough feel for performance (frame times) as you know. I noticed a significant delta when recreating them constantly, which is why I made it a habit to just resort to using long-lived ones. It is much faster to just empty the buffer and re-fill it constantly than always have to construct/destruct the object including its internal state etc.

Reply:
I don't know about the perf cost, never had to measure it. I guess one could always encapsulate this using a static variable in a function. Not saying you should do it, just a random thought.